lib/evm/operation/block_information.ex

```diff
@@ -9,42 +9,31 @@ defmodule EVM.Operation.BlockInformation do
 
   ## Examples
 
-      iex> block_b = %Block.Header{number: 2, parent_hash: "block_a"}
-      iex> block_a = %Block.Header{number: 1, parent_hash: "gen_block"}
-      iex> genesis_block = %Block.Header{number: 0, parent_hash: <<0x00::256>>}
-      iex> block_map = %{"gen_block" => genesis_block, "block_a" => block_a, "block_b" => block_b}
+      iex> block_b = %Block.Header{number: 2, mix_hash: "block_b"}
+      iex> block_a = %Block.Header{number: 1, mix_hash: "block_a"}
+      iex> genesis_block = %Block.Header{number: 0, mix_hash: <<0x00::256>>}
+      iex> block_map = %{<<0x00::256>> => genesis_block, "block_a" => block_a, "block_b" => block_b}
       iex> block_interface = EVM.Interface.Mock.MockBlockInterface.new(block_b, block_map)
       iex> exec_env = %EVM.ExecEnv{block_interface: block_interface}
       iex> EVM.Operation.BlockInformation.blockhash([3], %{exec_env: exec_env})
       0
       iex> EVM.Operation.BlockInformation.blockhash([2], %{exec_env: exec_env})
-      0
+      "block_b"
       iex> EVM.Operation.BlockInformation.blockhash([1], %{exec_env: exec_env})
       "block_a"
       iex> EVM.Operation.BlockInformation.blockhash([0], %{exec_env: exec_env})
-      "gen_block"
+      <<0::256>>
       iex> EVM.Operation.BlockInformation.blockhash([-1], %{exec_env: exec_env})
       0
   """
   @spec blockhash(Operation.stack_args, Operation.vm_map) :: Operation.op_result
   def blockhash([block_number], %{exec_env: exec_env}) do
```

Comment:
Hmm, I guess I missed the forest for the trees? Your implementation is significantly simpler. cool.